hs-ucan/library/Crypto/Key/Asymmetric/Public.hs

```diff
@@ -36,3 +46,43 @@ genRSA2048 = do
 
             Just private_qinv ->
               return RSA.PrivateKey {..}
+
+
+decodeASN1DERRSAPublicKey :: ByteString -> Either String RSA.PublicKey
+decodeASN1DERRSAPublicKey bs =
+  let
+    spki =
+      [ ASN1.Start ASN1.Sequence
+      , ASN1.Start ASN1.Sequence
+      , ASN1.OID [1,2,840,113549,1,1,1]
+      , ASN1.Null
+      , ASN1.End ASN1.Sequence
+      , ASN1.BitString (ASN1.toBitArray bs (BS.length bs * 8))
+      , ASN1.End ASN1.Sequence
+      ]
+  in case ASN1.fromASN1 spki of
+    Right (X509.PubKeyRSA pk, _) ->
+      Right pk
+
+    Right (pk, _) ->
+      Left $ "Couldn't parse RSAPublicKey (ASN1 DER encoded). Different format provided: " <> show pk
+
+    Left err ->
+      Left $ show err
+
+
+encodeASN1DERRSAPublicKey :: RSA.PublicKey -> ByteString
+encodeASN1DERRSAPublicKey pk =
+  case ASN1.toASN1 (X509.PubKeyRSA pk) [] of
+    [ ASN1.Start ASN1.Sequence
+      , ASN1.Start ASN1.Sequence
+      , ASN1.OID [1,2,840,113549,1,1,1]
+      , ASN1.Null
+      , ASN1.End ASN1.Sequence
+      , ASN1.BitString bitArray
+      , ASN1.End ASN1.Sequence
+      ] ->
+        ASN1.bitArrayGetData bitArray
```

Comment:
Not that it's a big problem, but kind of weird alignment here, no?

Reply:
(Lines 78-84)

Reply:
Yes, I agree. But for some very weird reason, it doesn't parse with the correct alignment :(
![image](https://user-images.githubusercontent.com/1430958/151787926-bad3c9b3-90fe-45d8-b143-e2eaa3b010d5.png)